### PR TITLE
cleanup of gnu find

### PR DESCRIPTION
**Pull request by rok-cesnovar**
#### Submisison Checklist

- [x] Run tests: `./runCmdStanTests.py src/test`
- [x] Declare copyright holder and open-source license: see below

#### Summary:

Continuing the work in stan-dev/math#1567 to close stan-dev/cmdstan#786

The issue is that our makefiles rely on the gnu find command that is not available on Windows.

stan-dev/math#1567 introduced a findfiles function that is a recursive version of the makefiles wildcard function.

This PR uses that in stan-dev/cmdstan

#### Copyright and Licensing

Please list the copyright holder for the work you are submitting (this will be you or your assignee, such as a university or company): Rok Češnovar

By submitting this pull request, the copyright holder is agreeing to license the submitted work under the following licenses:
- Code: BSD 3-clause (https://opensource.org/licenses/BSD-3-Clause)
- Documentation: CC-BY 4.0 (https://creativecommons.org/licenses/by/4.0/)

### CODE REVIEW

File: makefile [hunk 1]
```diff
@@ -218,9 +218,9 @@ clean: clean-manual
 
 clean-deps:
 	@echo '  removing dependency files'
-	$(shell find src $(STAN)src/stan $(MATH)stan -type f -name '*.d' -exec rm {} +)
-	$(shell find src $(STAN)src/stan $(MATH)stan -type f -name '*.d.*' -exec rm {} +)
-	$(shell find src $(STAN)src/stan $(MATH)stan -type f -name '*.dSYM' -exec rm {} +)
+	$(RM) $(call findfiles,src,*.d) $(call findfiles,src/stan,*.d) $(call findfiles,$(MATH)/stan,*.d)
+	$(RM) $(call findfiles,src,*.d.*) $(call findfiles,src/stan,*.d.*) $(call findfiles,$(MATH)/stan,*.d.*)
+	$(RM) $(call findfiles,src,*.dSYM) $(call findfiles,src/stan,*.dSYM) $(call findfiles,$(MATH)/stan,*.dSYM)
```
Comment (mitzimorris) [inline makefile]: when doing `make clean-all` console messages are a bunch of blank lines -  should be list of files that will be cleaned up.

**Comment by mitzimorris** [comment]
running `make build` on develop, followed by `make clean-all` 👍 
```
--- CmdStan v2.22.1 built ---
~/github/stan-dev/cmdstan (develop)> make clean-all
rm -f -r doc
cd src/docs/cmdstan-guide; rm -f *.brf *.aux *.bbl *.blg *.log *.toc *.pdf *.out *.idx *.ilg *.ind *.cb *.cb2 *.upa
  removing dependency files
  cleaning sundials targets
rm -f stan/lib/stan_math/lib/sundials_5.1.0/lib/libsundials_cvodes.a stan/lib/stan_math/lib/sundials_5.1.0/lib/libsundials_idas.a stan/lib/stan_math/lib/sundials_5.1.0/lib/libsundials_kinsol.a stan/lib/stan_math/lib/sundials_5.1.0/lib/libsundials_nvecserial.a stan/lib/stan_math/lib/sundials_5.1.0/src/cvodes/cvodea.o stan/lib/stan_math/lib/sundials_5.1.0/src/cvodes/cvodea_io.o stan/lib/stan_math/lib/sundials_5.1.0/src/cvodes/cvodes.o stan/lib/stan_math/lib/sundials_5.1.0/src/cvodes/cvodes_bandpre.o stan/lib/stan_math/lib/sundials_5.1.0/src/cvodes/cvodes_bbdpre.o stan/lib/stan_math/lib/sundials_5.1.0/src/cvodes/cvodes_diag.o stan/lib/stan_math/lib/sundials_5.1.0/src/cvodes/cvodes_direct.o stan/lib/stan_math/lib/sundials_5.1.0/src/cvodes/cvodes_io.o stan/lib/stan_math/lib/sundials_5.1.0/src/cvodes/cvodes_ls.o stan/lib/stan_math/lib/sundials_5.1.0/src/cvodes/cvodes_nls.o stan/lib/stan_math/lib/sundials_5.1.0/src/cvodes/cvodes_nls_sim.o stan/lib/stan_math/lib/sundials_5.1.0/src/cvodes/cvodes_nls_stg.o stan/lib/stan_math/lib/sundials_5.1.0/src/cvodes/cvodes_nls_stg1.o stan/lib/stan_math/lib/sundials_5.1.0/src/cvodes/cvodes_spils.o stan/lib/stan_math/lib/sundials_5.1.0/src/idas/idaa.o stan/lib/stan_math/lib/sundials_5.1.0/src/idas/idaa_io.o stan/lib/stan_math/lib/sundials_5.1.0/src/idas/idas.o stan/lib/stan_math/lib/sundials_5.1.0/src/idas/idas_bbdpre.o stan/lib/stan_math/lib/sundials_5.1.0/src/idas/idas_direct.o stan/lib/stan_math/lib/sundials_5.1.0/src/idas/idas_ic.o stan/lib/stan_math/lib/sundials_5.1.0/src/idas/idas_io.o stan/lib/stan_math/lib/sundials_5.1.0/src/idas/idas_ls.o stan/lib/stan_math/lib/sundials_5.1.0/src/idas/idas_nls.o stan/lib/stan_math/lib/sundials_5.1.0/src/idas/idas_nls_sim.o stan/lib/stan_math/lib/sundials_5.1.0/src/idas/idas_nls_stg.o stan/lib/stan_math/lib/sundials_5.1.0/src/idas/idas_spils.o stan/lib/stan_math/lib/sundials_5.1.0/src/kinsol/kinsol.o stan/lib/stan_math/lib/sundials_5.1.0/src/kinsol/kinsol_bbdpre.o stan/lib/stan_math/lib/sundials_5.1.0/src/kinsol/kinsol_direct.o stan/lib/stan_math/lib/sundials_5.1.0/src/kinsol/kinsol_io.o stan/lib/stan_math/lib/sundials_5.1.0/src/kinsol/kinsol_ls.o stan/lib/stan_math/lib/sundials_5.1.0/src/kinsol/kinsol_spils.o stan/lib/stan_math/lib/sundials_5.1.0/src/nvector/serial/nvector_serial.o stan/lib/stan_math/lib/sundials_5.1.0/src/sundials/sundials_band.o stan/lib/stan_math/lib/sundials_5.1.0/src/sundials/sundials_dense.o stan/lib/stan_math/lib/sundials_5.1.0/src/sundials/sundials_direct.o stan/lib/stan_math/lib/sundials_5.1.0/src/sundials/sundials_futils.o stan/lib/stan_math/lib/sundials_5.1.0/src/sundials/sundials_iterative.o stan/lib/stan_math/lib/sundials_5.1.0/src/sundials/sundials_linearsolver.o stan/lib/stan_math/lib/sundials_5.1.0/src/sundials/sundials_math.o stan/lib/stan_math/lib/sundials_5.1.0/src/sundials/sundials_matrix.o stan/lib/stan_math/lib/sundials_5.1.0/src/sundials/sundials_nonlinearsolver.o stan/lib/stan_math/lib/sundials_5.1.0/src/sundials/sundials_nvector.o stan/lib/stan_math/lib/sundials_5.1.0/src/sundials/sundials_nvector_senswrapper.o stan/lib/stan_math/lib/sundials_5.1.0/src/sundials/sundials_version.o stan/lib/stan_math/lib/sundials_5.1.0/src/sunlinsol/band/sunlinsol_band.o stan/lib/stan_math/lib/sundials_5.1.0/src/sunlinsol/dense/sunlinsol_dense.o stan/lib/stan_math/lib/sundials_5.1.0/src/sunmatrix/band/sunmatrix_band.o stan/lib/stan_math/lib/sundials_5.1.0/src/sunmatrix/dense/sunmatrix_dense.o stan/lib/stan_math/lib/sundials_5.1.0/src/sunnonlinsol/fixedpoint/sunnonlinsol_fixedpoint.o stan/lib/stan_math/lib/sundials_5.1.0/src/sunnonlinsol/newton/sunnonlinsol_newton.o
  cleaning Intel TBB targets
rm -f -rf stan/lib/stan_math/lib/tbb
rm -f -r test
rm -f 
rm -f 
rm -f 
rm -f 
rm -f bin/stanc bin/stanc2 bin/stansummary bin/print bin/diagnose
rm -f -r src/cmdstan/main.o bin/cmdstan
rm -f stan/src/stan/model/model_header.hpp.gch
rm -f examples/bernoulli/bernoulli examples/bernoulli/bernoulli.o examples/bernoulli/bernoulli.d examples/bernoulli/bernoulli.hpp
~/github/stan-dev/cmdstan (develop)> 
```

on branch `make_clean_gnu_find`  get too many blank lines so you lose track of work done:

```
--- CmdStan v2.22.1 built ---
~/github/stan-dev/cmdstan (make_clean_gnu_find)> make clean-all
rm -f -r doc
cd src/docs/cmdstan-guide; rm -f *.brf *.aux *.bbl *.blg *.log *.toc *.pdf *.out *.idx *.ilg *.ind *.cb *.cb2 *.upa
  removing dependency files
rm -f                                                                                                                                                                                                                src/cmdstan/stanc.d                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                
  cleaning sundials targets
rm -f stan/lib/stan_math/lib/sundials_5.1.0/lib/libsundials_cvodes.a stan/lib/stan_math/lib/sundials_5.1.0/lib/libsundials_idas.a stan/lib/stan_math/lib/sundials_5.1.0/lib/libsundials_kinsol.a stan/lib/stan_math/lib/sundials_5.1.0/lib/libsundials_nvecserial.a stan/lib/stan_math/lib/sundials_5.1.0/src/cvodes/cvodea.o stan/lib/stan_math/lib/sundials_5.1.0/src/cvodes/cvodea_io.o stan/lib/stan_math/lib/sundials_5.1.0/src/cvodes/cvodes.o stan/lib/stan_math/lib/sundials_5.1.0/src/cvodes/cvodes_bandpre.o stan/lib/stan_math/lib/sundials_5.1.0/src/cvodes/cvodes_bbdpre.o stan/lib/stan_math/lib/sundials_5.1.0/src/cvodes/cvodes_diag.o stan/lib/stan_math/lib/sundials_5.1.0/src/cvodes/cvodes_direct.o stan/lib/stan_math/lib/sundials_5.1.0/src/cvodes/cvodes_io.o stan/lib/stan_math/lib/sundials_5.1.0/src/cvodes/cvodes_ls.o stan/lib/stan_math/lib/sundials_5.1.0/src/cvodes/cvodes_nls.o stan/lib/stan_math/lib/sundials_5.1.0/src/cvodes/cvodes_nls_sim.o stan/lib/stan_math/lib/sundials_5.1.0/src/cvodes/cvodes_nls_stg.o stan/lib/stan_math/lib/sundials_5.1.0/src/cvodes/cvodes_nls_stg1.o stan/lib/stan_math/lib/sundials_5.1.0/src/cvodes/cvodes_spils.o stan/lib/stan_math/lib/sundials_5.1.0/src/idas/idaa.o stan/lib/stan_math/lib/sundials_5.1.0/src/idas/idaa_io.o stan/lib/stan_math/lib/sundials_5.1.0/src/idas/idas.o stan/lib/stan_math/lib/sundials_5.1.0/src/idas/idas_bbdpre.o stan/lib/stan_math/lib/sundials_5.1.0/src/idas/idas_direct.o stan/lib/stan_math/lib/sundials_5.1.0/src/idas/idas_ic.o stan/lib/stan_math/lib/sundials_5.1.0/src/idas/idas_io.o stan/lib/stan_math/lib/sundials_5.1.0/src/idas/idas_ls.o stan/lib/stan_math/lib/sundials_5.1.0/src/idas/idas_nls.o stan/lib/stan_math/lib/sundials_5.1.0/src/idas/idas_nls_sim.o stan/lib/stan_math/lib/sundials_5.1.0/src/idas/idas_nls_stg.o stan/lib/stan_math/lib/sundials_5.1.0/src/idas/idas_spils.o stan/lib/stan_math/lib/sundials_5.1.0/src/kinsol/kinsol.o stan/lib/stan_math/lib/sundials_5.1.0/src/kinsol/kinsol_bbdpre.o stan/lib/stan_math/lib/sundials_5.1.0/src/kinsol/kinsol_direct.o stan/lib/stan_math/lib/sundials_5.1.0/src/kinsol/kinsol_io.o stan/lib/stan_math/lib/sundials_5.1.0/src/kinsol/kinsol_ls.o stan/lib/stan_math/lib/sundials_5.1.0/src/kinsol/kinsol_spils.o stan/lib/stan_math/lib/sundials_5.1.0/src/nvector/serial/nvector_serial.o stan/lib/stan_math/lib/sundials_5.1.0/src/sundials/sundials_band.o stan/lib/stan_math/lib/sundials_5.1.0/src/sundials/sundials_dense.o stan/lib/stan_math/lib/sundials_5.1.0/src/sundials/sundials_direct.o stan/lib/stan_math/lib/sundials_5.1.0/src/sundials/sundials_futils.o stan/lib/stan_math/lib/sundials_5.1.0/src/sundials/sundials_iterative.o stan/lib/stan_math/lib/sundials_5.1.0/src/sundials/sundials_linearsolver.o stan/lib/stan_math/lib/sundials_5.1.0/src/sundials/sundials_math.o stan/lib/stan_math/lib/sundials_5.1.0/src/sundials/sundials_matrix.o stan/lib/stan_math/lib/sundials_5.1.0/src/sundials/sundials_nonlinearsolver.o stan/lib/stan_math/lib/sundials_5.1.0/src/sundials/sundials_nvector.o stan/lib/stan_math/lib/sundials_5.1.0/src/sundials/sundials_nvector_senswrapper.o stan/lib/stan_math/lib/sundials_5.1.0/src/sundials/sundials_version.o stan/lib/stan_math/lib/sundials_5.1.0/src/sunlinsol/band/sunlinsol_band.o stan/lib/stan_math/lib/sundials_5.1.0/src/sunlinsol/dense/sunlinsol_dense.o stan/lib/stan_math/lib/sundials_5.1.0/src/sunmatrix/band/sunmatrix_band.o stan/lib/stan_math/lib/sundials_5.1.0/src/sunmatrix/dense/sunmatrix_dense.o stan/lib/stan_math/lib/sundials_5.1.0/src/sunnonlinsol/fixedpoint/sunnonlinsol_fixedpoint.o stan/lib/stan_math/lib/sundials_5.1.0/src/sunnonlinsol/newton/sunnonlinsol_newton.o
rm -f                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                
rm -f                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                
  cleaning Intel TBB targets
rm -f -rf stan/lib/stan_math/lib/tbb
rm -f -r test
rm -f 
rm -f 
rm -f 
rm -f 
rm -f bin/stanc bin/stanc2 bin/stansummary bin/print bin/diagnose
rm -f -r src/cmdstan/main.o bin/cmdstan
rm -f stan/src/stan/model/model_header.hpp.gch
rm -f examples/bernoulli/bernoulli examples/bernoulli/bernoulli.o examples/bernoulli/bernoulli.d examples/bernoulli/bernoulli.hpp
~/github/stan-dev/cmdstan (make_clean_gnu_find)> 
```

**Comment by mitzimorris** [comment]
note number of blank lines in above output is approx - the selection from the terminal lost all blank lines so I had to edit them back in.

**Comment by rok-cesnovar** [comment]
Should be fixed by https://github.com/stan-dev/math/pull/1784

**Comment by mitzimorris** [comment]
@rok-cesnovar - in the future, please follow the branch naming conventions described here:   https://github.com/stan-dev/stan/wiki/Developer-process-overview#2-create-a-branch-for-the-issue

thanks!

**Comment by rok-cesnovar** [comment]
Deal! I keep forgetting.

**Pull request merged**